### DISCUSSION TITLE
VITS fails to synthesize very short or long text

### DISCUSSION THREAD

**Discussion by nicemanis** [coqui-ai/TTS#3451] [General Q&A]
I've been having difficulties with VITS training, so, to figure out where is the problem, I tried to train VITS on VCTK dataset from scratch and compare it with the pre-trained model, that comes with Coqui-TTS.

Here are audio samples from the pre-trained VITS VCTK model:

_One._

https://github.com/coqui-ai/TTS/assets/9463816/4bfae063-6f7b-44ee-bdb4-048fdfefe08f

_Hello world!_

https://github.com/coqui-ai/TTS/assets/9463816/65e04ea0-17a1-4d95-8d7c-d7dae994aa0b

_Quick brown fox jumps over the lazy dog._

https://github.com/coqui-ai/TTS/assets/9463816/fe7cf53e-dd4a-4758-9306-d02d00387cd1

_Every single Republican Senator voted against the measure, along with independent Senator Bernie Sanders, who had earlier in the day expressed reservations that the legislation included billions in military aid for Israel._

https://github.com/coqui-ai/TTS/assets/9463816/ebdd44e1-bdff-4041-85a1-70829b46a2a5

And here are audio samples from the model that I trained:

_One._

https://github.com/coqui-ai/TTS/assets/9463816/69cf6b9e-50e8-417b-9c9e-0c52e2002555

_Hello world!_

https://github.com/coqui-ai/TTS/assets/9463816/f134e0d9-6312-473c-94ca-8a3adacb4b66

_Quick brown fox jumps over the lazy dog._

https://github.com/coqui-ai/TTS/assets/9463816/0a4a2a75-8552-40d9-a975-b3fcdf06153a

_Every single Republican Senator voted against the measure, along with independent Senator Bernie Sanders, who had earlier in the day expressed reservations that the legislation included billions in military aid for Israel._

https://github.com/coqui-ai/TTS/assets/9463816/d119be4d-81a0-4156-ab1e-aedb97e9c04d

As you can hear, the pre-trained model sounds fine, but my model fails to synthesize very short or long sentences. It produces decent results on short/medium sentences (with some occasional stutters/pauses between words). I trained the model for 1M steps, same as the pre-trained checkpoint. I've already listed the config details in a previous discussion: https://github.com/coqui-ai/TTS/discussions/3377

My first suspicion was that there could be an issue with the `max_audio_len` and `min_audio_len` parameters in the config, which I had set to `10 * SAMPLE_RATE` and `1 * SAMPLE_RATE` respectively. Judging from the long sentence sample, it seems as if the model is trying to fit the whole sentence within the 10 second limit. I tried to set them to `float("inf")` and `1` during inference; however, it did not improve the result.

Has anyone encountered this issue or have any idea how to solve it? Maybe I'm missing some other parameters that should be changed during inference. I guess, the issue could be with the duration predictor. I'm using the stochastic duration predictor, same as the pre-trained model. I've also tried the deterministic duration predictor - it didn't have this issue on long texts, however the result was not ideal and had similar problems as reported here: https://github.com/coqui-ai/TTS/discussions/2834

**Comment by p0p4k** [comment]
your duration predictor seems to be an issue. How much was the final duration loss?

**Reply by nicemanis** [reply]
Here is the eval duration loss graph:
![image](https://github.com/coqui-ai/TTS/assets/9463816/c2d4da6f-d0db-435f-bbc2-9d7627c77942)

Here is the train duration loss:
![image](https://github.com/coqui-ai/TTS/assets/9463816/3f9fd6ff-a396-48bf-8690-46e37adacb2b)

**Reply by p0p4k** [reply]
Try to replace use_sdp as false in config and train. To debug if this is the real issue or not.

**Reply by nicemanis** [reply]
It seems that the issue does not persist with `use_sdp=false`, here are some examples after 70K steps of training from scratch:

https://github.com/coqui-ai/TTS/assets/9463816/044ce846-86eb-4188-bca4-b6fd5faf0ead


https://github.com/coqui-ai/TTS/assets/9463816/e1fa02ff-50a5-46ce-99e3-225357fc4dff


https://github.com/coqui-ai/TTS/assets/9463816/d3124ceb-91fa-4db7-83e6-2f4f605a8410


https://github.com/coqui-ai/TTS/assets/9463816/79d8c8b3-b1d5-450f-adab-b4395be8e71d

So, it seems that there is an issue with the SDP implementation?

**Reply by p0p4k** [reply]
Sdp can sometimes cause unnatural pronunciations.

**Reply by nicemanis** [reply]
Ok, but regardless of the naturalness of the pronunciation, SDP completely fails to synthesize long texts, while there are no issues with DDP.

The main point is - I'm trying to reproduce the same results as the pre-trained Coqui-TTS VITS VCTK model. I'm using the same dataset and the recipe that is provided by the repo, yet I'm not able to get the same results. Is it possible to get the original training script or logs that it was trained on?

There must be a problem somewhere, or I'm not aware of some other differences that could influence the result. I will try to train the model using an older version of the repo.

**Reply by nicemanis** [reply]
While investigating duration predictor, I noticed incosistencies with the speakers:

https://github.com/coqui-ai/TTS/assets/9463816/0f8de41b-4979-4e58-8be7-32ad12ba8958


https://github.com/coqui-ai/TTS/assets/9463816/9811a505-6f9b-4fd1-bcbb-e0a838f3530e


https://github.com/coqui-ai/TTS/assets/9463816/358ef059-513a-4d18-a789-6a362483ddad

I tried synthesizing with the pre-trained VITS VCTK model using "p277" speaker, and, as you can hear in the samples, the voice sounds different from the reference `p277` sample. I was wondering if this could cause issues when fine-tuning. Also, duration predictor is conditioned on speaker embedding

**Reply by eginhard** [reply]
See #2258
